### ISSUE TITLE
[Enhancement]: Unified libraries

### ISSUE THREAD

**Issue by dedors** [advplyr/audiobookshelf#2236]
### Describe the feature/enhancement

It might not be a high-priority item, but I'd like to suggest the idea for future consideration, gathering additional ideas and consider it when planning related structures:

I have a large number of books organized into four separate libraries based on folder structure, each with different structure requirements (e.g. series with hunderds of books vs single). 
A unified bookshelf that includes all libraries would offer benefits, such as continue listening and recently added items. 
It could provide a search across all libraries.
Additionally, it could unify collections (e.g. 'Listen Later' and 'Favorites'), but that would require additional changes for collection (see more below).

### Describe the solution you'd like
Add a new entry at the top of the library dropdown list called 'All libraries.' 
Consider including an option in the library settings page for users who prefer not to have this.

### Describe alternatives you've considered
I also thought about having a separate row for 'Listen Later' and 'Favorites' in the bookshelf. If this becomes a future feature in the collection settings, it could include an option to choose between 'This Library', 'Unified Library' and 'All Libraries' (or even selectable libraries). The latter two would allow collections like 'Favorites' and 'Listen Later' to be global, and the 'All Libraries' option would show even if your in a specific library (e.g. no matter what library you have selected, it will always show favorites from all library).

**Comment by iconoclasthero** [comment]
Yes, this is something that I see being useful.  I tl;dr'ed the OP, but pursuant to my last suggestion (https://github.com/advplyr/audiobookshelf/issues/2350), it would be ideal if there was some sort of global search that would obviously be facilitated by having a global/unified library.

**Comment by Flameingskull** [comment]
I'd love to see this as an enhancement for the Home page so I can see recently added books and books to continue etc. from multiple large libraries. I made a separate enhancement ticket without realising this existed so I'll just add my vote to this one

**Comment by Mancolt** [comment]
Yes, I agree. Adding my vote here too. Don't understand why this wouldn't be useful, especially for less tech savvy users who don't want/need the library distinction and just want to see all books they have access to.
